python/src/aiconfig/editor/client/src/components/prompt/prompt_input/schema_renderer/PromptInputAttachmentsSchemaRenderer.tsx

```diff
@@ -106,18 +114,18 @@ export default memo(function PromptInputAttachmentsSchemaRenderer({
       {/* If the schema supports more attachments, add another attachment input below existing attachments. This
        * will allow the user to add more attachments without 'clearing' existing ones.
        */}
-      {schema.max_items == null ||
-        (numAttachments < schema.max_items && (
-          <EditableAttachmentRenderer
-            schema={schema}
-            onUpdateAttachment={(attachment) =>
-              onUpdateAttachment(attachment, numAttachments + 1)
-            }
-            onAddAttachments={(addedAttachments) =>
-              onAddAttachments(addedAttachments, numAttachments + 1)
-            }
-          />
-        ))}
+      {(schema.max_items == null || numAttachments < schema.max_items) && (
+        <EditableAttachmentRenderer
+          key={numAttachments}
```

Comment:
shouldn't the key be something unique? Or is this created once and never changed?

Reply:
Ah, let me add a comment. This is actually kind of a react trick to force each new EditableAttachmentRenderer to be freshly mounted instead of maintaining state with its previous render (because the attachment now exists and is rendered above)